### PR TITLE
adding option to install calico with kubernetes data storage

### PR DESCRIPTION
**Pull request by wzhliang**
Wrote this a few weeks back but didn't have time to send the PR. Saw a request today and decided this might be helpful. Please review.

### CODE REVIEW

File: okdc-centos.sh [hunk 1]
```diff
@@ -150,6 +164,9 @@ install_network() {
 				install_calico_with_etcd
 				;;
 			3)
+				install_calico_with_kdd
```
Comment (ledzep2) [inline okdc-centos.sh:170]: Can you also add the handling for noinput mode? Just add another case in the switch above this one.

File: okdc-centos.sh [hunk 2]
```diff
@@ -227,7 +244,7 @@ update_kubelet() {
 	Wants=flexv.service
 	After=flexv.service
 	[Service]
-	Environment="KUBELET_NETWORK_ARGS=--network-plugin=kubenet"
+	Environment="KUBELET_NETWORK_ARGS=--network-plugin=cni"
```
Comment (ledzep2) [inline okdc-centos.sh:250]: Have you tested this one on kubeadm 1.7? This used to lead to a deadlock while kubeadm init. Not sure if it's fixed. The previous method is to wait until kubeadm is done then remove this line in `patch_kubelet`.

**Review by ledzep2 — CHANGES_REQUESTED** [review]

File: okdc-centos.sh (hunk 1 above)
Comment (wzhliang) [reply okdc-centos.sh:170]: ok

File: okdc-centos.sh (hunk 2 above)
Comment (wzhliang) [reply okdc-centos.sh:250]: Yes. I was able to install 1.7.3 with my script

File: okdc-centos.sh (hunk 1 above)
Comment (ledzep2) [reply okdc-centos.sh:170]: Did you add it?

Comment (wzhliang) [reply okdc-centos.sh:170]: I added NOINPUT envar handling like the other cases. Not sure if that's what you wanted.

Comment (ledzep2) [reply okdc-centos.sh:170]: Oh, no. There's another case switch (line #137) just above this one which runs when $Network is predefined or in NOINPUT mode. We will need a calico_kdd (or something like that) option added so users can use that in NOINPUT mode.

Comment (wzhliang) [reply okdc-centos.sh:170]: done

Comment (ledzep2) [reply okdc-centos.sh:170]: lgtm. Please squash and we will get this merged.

**Review by ledzep2 — APPROVED** [review]

**Pull request closed without merging**